store/copr/region_cache.go

```diff
@@ -77,56 +77,43 @@ func (l *LocationKeyRanges) splitKeyRangesByBuckets() []*LocationKeyRanges {
 	loc := l.Location
 	res := []*LocationKeyRanges{}
 	for ranges.Len() > 0 {
-		bucket := loc.LocateBucket(ranges.At(0).StartKey)
-		if bucket == nil {
-			// TODO(youjiali1995): if it's overlapped with some buckets, it can be splitted.
-			//
-			// Buckets information may not be up-to-date and accurate.
-			// Find all ranges that can't be located in a bucket and make it one task.
-			i := 1
-			for ; i < ranges.Len(); i++ {
-				if loc.LocateBucket(ranges.At(i).StartKey) != nil {
-					break
-				}
-			}
-			res = append(res, &LocationKeyRanges{l.Location, ranges.Slice(0, i)})
-			ranges = ranges.Slice(i, ranges.Len())
-		} else {
-			// Iterate to the first range that is not complete in the bucket.
-			var r kv.KeyRange
-			var i int
-			for ; i < ranges.Len(); i++ {
-				r = ranges.At(i)
-				if !(bucket.Contains(r.EndKey) || bytes.Equal(bucket.EndKey, r.EndKey)) {
-					break
-				}
-			}
-			// All rest ranges belong to the same bucket.
-			if i == ranges.Len() {
-				res = append(res, &LocationKeyRanges{l.Location, ranges})
+		// ranges must be in loc.region, so the bucket returned by loc.LocateBucketV2 is guaranteed to be not nil
+		bucket := loc.LocateBucketV2(ranges.At(0).StartKey)
```

Comment:
What does the `V2` mean? Do we need to use `V1` somewhere else? If not we could remove the version in the name and try to avoid misunderstranding.